internal/data/transactions.go

```diff
@@ -18,6 +18,98 @@ type TransactionModel struct {
 	MetricsService metrics.MetricsService
 }
 
+func (m *TransactionModel) GetByHash(ctx context.Context, hash string) (*types.Transaction, error) {
+	const query = `SELECT * FROM transactions WHERE hash = $1`
```

Comment:
❓ If the client makes a query that only requests the hash, aren't we taking a performance hit by fetching all fields? Can we only fetch the columns necessary to fulfill the request?

Reply:
@JakeUrban This query is not to get the hash but to get the txn given a hash. So from the graphql perspective, users query the txn by hash and then can request any field on the transaction which will be resolved using the resolvers

Reply:
Ok I think this is fine for now, but the performance cost of fetching all rows isn't negligible when our tables have large text fields for `metaXdr` for example. Or when tables have a high number of optional columns like statechanges, we should probably only query for the columns associated with the classification/type requested (if the client specified a type or set of types.

Reply:
@JakeUrban Ok, let me think about this problem. I still think it would not be a performance hit to just fetch the columns but will do some research on my end.

The issue occurs specifically when people return the child objects too in the query - txn's operations, state changes etc... which we are not doing here. But for fetching of columns, I am not exactly sure.